src/ui/screens/AddBountyScreen.tsx

```diff
@@ -0,0 +1,182 @@
+  const [bountyAllocation, SetBountyAllocation] = React.useState('');
+
+  const [account, setAccount] = React.useState<Account>();
+  const modelClose = () => modalRef.current?.close();
+  const startTx = useApiTx();
+  const {api} = useApi();
+  const snackbar = useSnackbar();
+
+  const {calculatedBountyBond, bountyAllocationBN, isBountyValid, isBountyTitleValid} = useMemo(() => {
+    if (bounty && bounty.bountyDepositBase) {
+      const calculatedBountyBond = calculateBountyBond(
+        bountyTitle,
+        formattedStringToBn(bounty.bountyDepositBase),
+        formattedStringToBn(bounty.dataDepositPerByte),
+      );
+      const bountyAllocationBN = stringToBn(bountyAllocation);
+      const isBountyValid = bountyAllocationBN
+        ? bountyAllocationBN.gt(formattedStringToBn(bounty.bountyValueMinimum))
+        : false;
+      const isBountyTitleValid =
+        Number(bountyTitle.length) <= Number(bounty.maximumReasonLength) && Number(bountyTitle.length) > 0;
+      return {
+        calculatedBountyBond,
+        bountyAllocationBN,
+        isBountyValid,
+        isBountyTitleValid,
+      };
+    }
+    return {
+      calculatedBountyBond: formattedStringToBn(bounty?.bountyDepositBase),
+      bountyAllocationBN: stringToBn(bountyAllocation),
+      isBountyValid: false,
+      isBountyTitleValid: bountyTitle.length < 1,
+    };
+  }, [bountyTitle, bountyAllocation, bounty, stringToBn]);
+
+  const disabled = !bountyAllocation || !account || !isBountyValid || !isBountyTitleValid;
+
+  const submitBounty = () => {
+    if (account && api && bountyAllocationBN) {
+      startTx({
+        address: account.address,
+        txMethod: `${api.tx.bounties ? `bounties.proposeBounty` : `treasury.proposeBounty`}`,
+        params: [bountyAllocationBN, bountyTitle],
+      })
+        .then(() => {
+          snackbar('New bounty created');
+        })
+        .catch(() => {
+          snackbar('Error while submitting bounty');
+        });
+      modalRef.current?.close();
+    }
+  };
+
+  return (
+    <Modalize ref={modalRef} adjustToContentHeight onClose={navigation.goBack} panGestureEnabled={false}>
+      <Layout style={styles.container}>
+        <View style={globalStyles.alignCenter}>
+          <Headline>{'Add Bounty'}</Headline>
+        </View>
+        <View style={styles.container}>
+          <InputLabel label={'Bounty Title:'} helperText={'Description of the Bounty (to be stored on-chain)'} />
+          <TextInput
+            dense
+            multiline
+            numberOfLines={4}
+            autoComplete="off"
+            mode="outlined"
+            placeholder="Enter bounty title"
+            value={bountyTitle}
+            onChangeText={(value) => setBountyTitle(value)}
+          />
+          {!isBountyTitleValid && bountyTitle.length !== 0 ? (
+            <HelperText type="error">{`Exceeding maximum reasoning length`}</HelperText>
+          ) : null}
```

Comment:
No need to wrap it with the condition. The `<HelperText />` component has a `visible` prop that you can use.
```suggestion
          <HelperText type="error" visible={!isBountyTitleValid && bountyTitle.length !== 0}>{`Exceeding maximum reasoning length`}</HelperText>
```

Reply:
The reason I wrapped the helper is that it won't add a new line in the UI. it is mostly to hide the text line. please check with the image to see the difference:
Without `<HelperText/>` wrapped:
![Simulator Screen Shot - iPhone 12 - 2022-03-17 at 06 18 59](https://user-images.githubusercontent.com/22363352/158742556-f6114912-9f8c-4299-9944-a976fa019269.png)

with `<HelperText/>` wrapped:
![Simulator Screen Shot - iPhone 12 - 2022-03-17 at 06 19 11](https://user-images.githubusercontent.com/22363352/158742597-5a3f64e6-2e12-49b3-857c-00875a97ac76.png)